### PR TITLE
Search mock data

### PR DESCRIPTION
**Pull request by michael-brennan2005**
Adds club description + club tag generation via ChatGPT. 

Some notes:
* I rewrote the club scraper to a single python script (sorry @garrettladley). Club scraping is a one-off, non resource intensive task with a lot of string and API fiddling; ultimately I think it makes much more sense for this to be a python script than a Rust project, especially considering that this script is less LoC (470 lines of rust -> 226 lines of python), and that it was faster to write this script from scratch than it was to add the feature to the existing Rust project (Rust ownership semantic hell).
* ChatGPT will generate invalid JSON for club description prompts very rarely (only 2 out of ~40 on one run) and invalid JSON for club tags a lot (half of the tags it generates are ones that weren't given as tags to use).  If this is something we want to reduce w/ prompt engineering or something we can, but its very rare for club descriptions and only 50% of tags being valid is still a lot of tags (~1400 for ~600 clubs on one run), so I think its OK for what this data is intended for. 
* Using ChatGPT to generate this data is very slow - since prompts can't be batched and thus only one description or one set of tags can be made per request - and very expensive - during making this script I used $0.66 of credits, which is 11x more than the credits used for all of the embeddings we've made from the start of this project to now. This is why I've commited the mock.sql, so people can just use that as provided, and we only regenerate the mock.sql if the script changes. Trying to batch the descriptions and/or tag creation through prompt engineering is possible, but, given that this is a one-off script and that we're running into invalid JSON for very simple requests (the description and set of tags generation), imo that isn't something that's worth spending time on.

### CODE REVIEW

File: .github/workflows/mock_data.yml
Comment (garrettladley) [inline .github/workflows/mock_data.yml:1]: we should still have CI for this, I'll open an issue and work on that

File: frontend/sac-mobile/expo-env.d.ts
Comment (garrettladley) [inline frontend/sac-mobile/expo-env.d.ts:1]: relic from old frontend directory organization. can we remove?

File: mock_data/main.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    ]
+}
+
+TAGS_STRING = "" # for tags_prompt
+for key, value in MOCK_CATEGORIES_AND_TAGS.items():
+    for tag in value:
+        TAGS_STRING += f"\"{tag}\", "
+TAGS_STRING = TAGS_STRING[:-2] # remove the last ", "
+
+def description_prompt(name, preview):
```
Comment (garrettladley) [inline mock_data/main.py:90]: i'm a big type hint head, please forgive these nits

```suggestion
def description_prompt(name: str, preview: str) -> str:
```

File: mock_data/main.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    return f"""
+    I need your help writing club descriptions to create some mock data for an app I'm working on. The club's name is \"{name}\", and its short summary is \"{preview}\".
+    Generate a one to two paragraph description of what the club is and what it does. Use a mix of friendly, welcoming, and informative language and content. Store in this a JSON field \"description\".
+    Your output should just be the JSON with the generated description, no other text. Thank you!
+    """
+
+def tags_prompt(name, description):
```
Comment (garrettladley) [inline mock_data/main.py:97]:
```suggestion
def tags_prompt(name: str, description: str) -> str:
```

File: mock_data/main.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        "recruitment_type": random.choice(["unrestricted", "application"]),
+        "tags": tags
+    })
+
+    bar.next()
+bar.finish()
+print(f"Bad club descriptions (JSON not created properly): {bad_club_descriptions}")
+
+# Generate the UUIDS for Categories, Tags, and a table for tags to categories.
+category_uuids = {}
```
Comment (garrettladley) [inline mock_data/main.py:166]:
```suggestion
category_uuids = {category: str(uuid4()) for category in MOCK_CATEGORIES_AND_TAGS}
```

File: mock_data/main.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+for category in MOCK_CATEGORIES_AND_TAGS:
```
Comment (garrettladley) [inline mock_data/main.py:167]:
```suggestion
```

File: mock_data/main.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    category_uuids[category] = str(uuid4())
```
Comment (garrettladley) [inline mock_data/main.py:168]:
```suggestion
```

File: mock_data/main.py [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+tag_uuids = {}
```
Comment (garrettladley) [inline mock_data/main.py:170]:
```suggestion
tag_uuids = {tag: str(uuid4()) for tag in MOCK_CATEGORIES_AND_TAGS.values() for tag in tags}
```

File: mock_data/main.py [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+tags_to_categories = {}
```
Comment (garrettladley) [inline mock_data/main.py:171]:
```suggestion
tags_to_categories = {tag: category for category, tags in MOCK_CATEGORIES_AND_TAGS.items() for tag in tags}
```

File: mock_data/main.py [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+for category, tags in MOCK_CATEGORIES_AND_TAGS.items():
```
Comment (garrettladley) [inline mock_data/main.py:172]:
```suggestion
```

File: mock_data/main.py [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+    for tag in tags:
```
Comment (garrettladley) [inline mock_data/main.py:173]:
```suggestion
```

File: mock_data/main.py [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+        tag_uuids[tag] = str(uuid4())
```
Comment (garrettladley) [inline mock_data/main.py:174]:
```suggestion
```

File: mock_data/main.py [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+        tags_to_categories[tag] = category
```
Comment (garrettladley) [inline mock_data/main.py:175]:
```suggestion
```

**Review by garrettladley — COMMENTED** [review]
nitty nitty pythonista changes, great work @michael-brennan2005 ! 🌞

File: mock_data/main.py [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+    ],
+    "MediaAndCommunication": [
+        "Journalism",
+        "Broadcasting",
+        "Film",
+        "PublicRelations",
+    ]
+}
+
+TAGS_STRING = "" # for tags_prompt
```
Comment (garrettladley) [inline mock_data/main.py:84]:
```suggestion
TAGS_STRING = ", ".join(f"\"{tag}\"" for tag in sum(MOCK_CATEGORIES_AND_TAGS.values(), []))
```

File: mock_data/main.py [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+for key, value in MOCK_CATEGORIES_AND_TAGS.items():
```
Comment (garrettladley) [inline mock_data/main.py:85]:
```suggestion
```

File: mock_data/main.py [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+    for tag in value:
```
Comment (garrettladley) [inline mock_data/main.py:86]:
```suggestion
```

File: mock_data/main.py [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+        TAGS_STRING += f"\"{tag}\", "
```
Comment (garrettladley) [inline mock_data/main.py:87]:
```suggestion
```

File: mock_data/main.py [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+TAGS_STRING = TAGS_STRING[:-2] # remove the last ", "
```
Comment (garrettladley) [inline mock_data/main.py:88]:
```suggestion
```

**Comment by michael-brennan2005** [comment]
@garrettladley Done

**Pull request merged**